context.go

```diff
@@ -228,6 +228,10 @@ func (c *Context) Error(err error) *Error {
 // Set is used to store a new key/value pair exclusively for this context.
 // It also lazy initializes  c.Keys if it was not used previously.
 func (c *Context) Set(key string, value interface{}) {
+	if c.KeysMutex == nil {
+		c.KeysMutex = &sync.RWMutex{}
+	}
```

Comment:
@appleboy @thinkerou Isn't the mutex creation still a race condition? For two threads both calling "Set" at the same time, the mutex could get created twice -- once for each thread. What about using "sync.once" here?

Reply:
Good catch.

Reply:
@felix-h fix in #2351